### ISSUE TITLE
[BUG] Tuplex crashes in exception resolver when take/collect() is called twice

### ISSUE THREAD

**Issue by KorlaMarch** [tuplex/tuplex#94]
The tuplex crashes from a corrupted exception partition when the pipeline is executed twice. This could be an issue from invalidating partition.

Example pipeline when it crashes:
```
ds = c.parallelize([(1, "A"),(2, "a"),(3, 2)]).filter(lambda a, b: a > 1)
ds.collect()
ds.collect()
```

**Comment by KorlaMarch** [comment]
@bgivertz

**Comment by LeonhardFS** [comment]
@bgivertz is this issue also present on your branch?

**Comment by LeonhardFS** [comment]
fallback partitions were not marked as immortal. Therefore, the 2nd call to `.collect()` accessed already invalidated partitions. Fixed in #112.
